### PR TITLE
CASMINST-4740 CODEOWNERS

### PR DESCRIPTION
**Pull request by rustydb**
Taking a stab at adding a codeowners.

This was created initially by @mitcharf and @rustydb.

Once this PR has approvals and has addressed any feedback backports will be created.

Backports:
- https://github.com/Cray-HPE/docs-csm/pull/1829
- https://github.com/Cray-HPE/docs-csm/pull/1830

### CODE REVIEW

File: .github/CODEOWNERS
Comment (mharding-hpe) [inline .github/CODEOWNERS:5]: Found the other owner we should use for that one

```suggestion
operations/CSM_product_management                       @Cray-HPE/cms-core-cfs
operations/CSM_product_management                       @Cray-HPE/cms-core-csm-product
```

**Review by mharding-hpe — APPROVED** [review]

File: .github/CODEOWNERS
Comment (johren-hpe) [inline .github/CODEOWNERS:14]: Need to add write permission for all of these new groups to get rid of the errors above.

Comment (johren-hpe) [inline .github/CODEOWNERS:14]: It's best to have a slash at the beginning of each of these lines.  That will make it specifically for the operations directory (for example) at the root of the repository.  Otherwise it will be for any operations directory in docs-csm.

**Review by johren-hpe — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS
Comment (rustydb) [reply .github/CODEOWNERS:14]: Yes, I was waiting until this had approvals as to not add teams unnecessarily. After this has approvals I'll also make Backports

**Review by johren-hpe — APPROVED** [review]

File: .github/CODEOWNERS
Comment (rustydb) [reply .github/CODEOWNERS]: Done.

**Pull request merged**
